python/pyarrow/tests/parquet/test_basic.py

```diff
@@ -63,21 +62,20 @@ def test_parquet_invalid_version(tempdir):
                      data_page_version="2.2")
 
 
-@parametrize_legacy_dataset
-def test_set_data_page_size(use_legacy_dataset):
+@pytest.mark.dataset
```

Comment:
I know this mark was in practice used through `@parametrize_legacy_dataset`, but we might not need to add it for all cases. Here, this `_check_roundtrip` is using `pq.read_table` under the hood. But for many cases (reading simple files), we have a fallback in `read_table` to use ParquetFile in case the dataset API is not available.

So I think we only need to mark those tests with `@pytest.mark.dataset` if it is actually testing a feature that is only supported in the dataset-based code path (eg reading multiple files instead of a single file, or when specifying a filter or partitioning)